contracts/JBERC20PaymentTerminal3_2.sol

```diff
@@ -64,13 +103,104 @@ contract JBERC20PaymentTerminal3_2 is JBPayoutRedemptionPaymentTerminal3_2 {
     )
   // solhint-disable-next-line no-empty-blocks
   {
+    PERMIT2 = _permit2;
+  }
 
+  //*********************************************************************//
+  // ----------------------- public transactions ----------------------- //
+  //*********************************************************************//
+
+  /// @notice Contribute tokens to a project and sets an allowance for this terminal (using Permit2).
+  /// @param _projectId The ID of the project being paid.
+  /// @param _amount The amount of terminal tokens being received, as a fixed point number with the same amount of decimals as this terminal. If this terminal's token is ETH, this is ignored and msg.value is used in its place.
+  /// @param _token The token being paid. This terminal ignores this property since it only manages one token.
+  /// @param _beneficiary The address to mint tokens for and pass along to the funding cycle's data source and delegate.
+  /// @param _minReturnedTokens The minimum number of project tokens expected in return, as a fixed point number with the same amount of decimals as this terminal.
+  /// @param _preferClaimedTokens A flag indicating whether the request prefers to mint project tokens into the beneficiaries wallet rather than leaving them unclaimed. This is only possible if the project has an attached token contract. Leaving them unclaimed saves gas.
+  /// @param _memo A memo to pass along to the emitted event, and passed along the the funding cycle's data source and delegate.  A data source can alter the memo before emitting in the event and forwarding to the delegate.
+  /// @param _metadata Bytes to send along to the data source, delegate, and emitted event, if provided.
+  /// @param _allowance The allowance to set for this terminal (using Permit2).
+  /// @return The number of tokens minted for the beneficiary, as a fixed point number with 18 decimals.
+  function payAndSetAllowance(
```

Comment:
@simplemachine92 I don't see your changes here that uses the permit2 transfer, and calls internal _pay instead of external pay. is this correct?

Reply:
tests pass so maybe we good to go